cupy/linalg/norms.py

```diff
@@ -241,11 +241,15 @@ def slogdet(a):
     util._assert_nd_squareness(a)
 
     dtype = numpy.promote_types(a.dtype.char, 'f')
-    real_dtype = dtype
+    if dtype == numpy.complex64:
+        real_dtype = numpy.dtype("f")
```

Comment:
I suppose if you have already known the dtype, just use the explicit form:
```suggestion
        real_dtype = numpy.float32
```

Reply:
How about writing as follows?
```py
real_dtype = numpy.dtype(dtype.char.lower())
```

Reply:
Ah, it's better. No more branching!

Reply:
Thank you!